gitlab/v4/objects/epics.py

```diff
@@ -42,11 +44,17 @@ class GroupEpicManager(CRUDMixin, RESTManager):
     )
     _types = {"labels": types.ListAttribute}
 
+    def get(self, id: Union[str, int], lazy: bool = False, **kwargs: Any) -> GroupEpic:
+        return cast(GroupEpic, super().get(id=id, lazy=lazy, **kwargs))
+
 
 class GroupEpicIssue(ObjectDeleteMixin, SaveMixin, RESTObject):
     _id_attr = "epic_issue_id"
+    # Define type for 'manager' here So mypy won't complain about
+    # 'self.manager.update()' call in the 'save' method.
+    manager: "GroupEpicIssueManager"
```

Comment:
We will hit this in a few more places it seems:

```
nejc@nuc:~/repos/python-gitlab$ grep -r 'self.manager.update' gitlab/v4
gitlab/v4/objects/labels.py:        server_data = self.manager.update(None, updated_data, **kwargs)
gitlab/v4/objects/labels.py:        server_data = self.manager.update(None, updated_data, **kwargs)
gitlab/v4/objects/epics.py:    # 'self.manager.update()' call in the 'save' method.
gitlab/v4/objects/epics.py:        self.manager.update(obj_id, updated_data, **kwargs)
```
Can we solve this with Protocols? Again, I haven't looked into it enough to give smart suggestions :grin: 
https://mypy.readthedocs.io/en/latest/more_types.html#advanced-uses-of-self-types

If we have to conditionally import `typing_extensions` I think it'd still be worth it if solves the issue.

Reply:
Protocols were added in Python 3.8. So I guess in another year we could do it after we deprecate Python 3.7. 😏

But to answer your question I am not sure if Protocols will or won't work. Since we are using a `Mixin` based design pattern I'm not sure.

Reply:
I don't think 3.8 is strictly necessary as there is https://pypi.org/project/typing-extensions/, but agreed needs more exploration and I see you already added them 😁